examples/name-service/programs/name-service/src/lib.rs

```diff
@@ -258,4 +99,37 @@ pub struct NameService<'info> {
     /// CHECK: Checked in light-system-program.
     #[authority]
     pub cpi_signer: AccountInfo<'info>,
+
+    #[light_account(init, seeds = [b"name-service", record.name.as_bytes()])]
+    pub record: LightAccount<NameRecord>,
+}
+
+#[light_accounts]
+pub struct UpdateRecord<'info> {
+    #[account(mut)]
+    #[fee_payer]
+    pub signer: Signer<'info>,
+    #[self_program]
+    pub self_program: Program<'info, crate::program::NameService>,
+    /// CHECK: Checked in light-system-program.
+    #[authority]
+    pub cpi_signer: AccountInfo<'info>,
+
+    #[light_account(mut, seeds = [b"name-service", record.name.as_bytes()])]
+    pub record: LightAccount<NameRecord>,
+}
+
+#[light_accounts]
+pub struct DeleteRecord<'info> {
+    #[account(mut)]
+    #[fee_payer]
+    pub signer: Signer<'info>,
+    #[self_program]
+    pub self_program: Program<'info, crate::program::NameService>,
```

Comment:
can we generate this in the macro since program `Program<'info, crate::program::NameService>,` is completely deterministic?